### ISSUE TITLE
🐛 [firebase_core] Cannot build Flutter project that using firebase_core v0.5.0 for iOS

### ISSUE THREAD

**Issue by DragonCherry** [FirebaseExtended/flutterfire#3986]
## Bug report

**Describe the bug**
Could not build Flutter app with dependency > firebase_core v0.5.0+1

### Steps to reproduce

1. Create Flutter project named 'fireabase_test_app'
2. Create and configure new firebase project and download GoogleService-Info.plist file, add it to folder 'firebase_test_app/Runner/' and add it to target Runner.
3. add firebase_core: ^0.5.0+1 on pubspec.yaml file.
4. flutter pug get
5. change main.dart file with firebase test app code
```
import 'dart:async';

import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  debugPaintSizeEnabled = false;
  runApp(App());
}

class App extends StatelessWidget {
  // Create the initialization Future outside of `build`:
  final Future<FirebaseApp> _initialization = Firebase.initializeApp();

  @override
  Widget build(BuildContext context) {
    return FutureBuilder(
      // Initialize FlutterFire:
      future: _initialization,
      builder: (context, snapshot) {
        // Check for errors
        if (snapshot.hasError) {
          return Text('Error');
        }

        // Once complete, show your application
        if (snapshot.connectionState == ConnectionState.done) {
          return Text('Success?');
        }

        // Otherwise, show something whilst waiting for initialization to complete
        return Center(child: CircularProgressIndicator());
      },
    );
  }
}


```
6. Debug run it, after Xcode build success, console output shows like this
```
Launching lib/main.dart on Yong's iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: GD2PMHGGVD
Running Xcode build...
Xcode build done.                                           27.6s
Failed to build iOS app
Error output from Xcode build:
↳
    2020-10-29 20:34:58.718 xcodebuild[30241:2802553] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7fd67037c870), Yong's iPhone, iPhone, 14.1 (18A8395), 405b4fbe495ec40cdc6fe309c9de096e883df0a2> == Underlying device preparation errors ==
    2020-10-29 20:34:58.718 xcodebuild[30241:2802553] [MT] iPhoneConnect: Failed _shouldMakeReadyForDevelopment check even though device is not locked by passcode.
    Domain: com.apple.platform.iphoneos
    Code: 5
    Failure Reason: allowsSecureServices: 1. isConnected: 0. Platform: <DVTPlatform:0x7fd66b533fa0:'com.apple.platform.iphoneos':<DVTFilePath:0x7fd66b533e20:'/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform'>>. DTDKDeviceIdentifierIsIDID: 0
    User Info: {
        DVTDeviceDescription = "\Ud83d\Udcf1<DVTiOSDevice (0x7fd67037c870), Yong's iPhone, iPhone, 14.1 (18A8395), 405b4fbe495ec40cdc6fe309c9de096e883df0a2>";
    }
    --
    2020-10-29 20:34:58.718 xcodebuild[30241:2802553] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7fd67037c870), Yong's iPhone, iPhone, 14.1 (18A8395), 405b4fbe495ec40cdc6fe309c9de096e883df0a2> == END: Underlying device preparation errors ==
    ** BUILD FAILED **


Xcode's output:
↳
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:91:5: Error: Expected a class member, but got 'assert'.
        assert(options != null,
        ^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:91:11: Error: Expected an identifier, but got '('.
    Try inserting an identifier before '('.
        assert(options != null,
              ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:91:20: Error: Expected ')' before this.
        assert(options != null,
                       ^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:95:5: Error: Expected a class member, but got 'if'.
        if (appInstances.containsKey(name)) {
        ^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:95:8: Error: Expected an identifier, but got '('.
    Try inserting an identifier before '('.
        if (appInstances.containsKey(name)) {
           ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:95:33: Error: Expected an identifier, but got '('.
    Try inserting an identifier before '('.
        if (appInstances.containsKey(name)) {
                                    ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:99:48: Error: Expected ')' before this.
        _initializeFirebaseAppFromMap(await channel.invokeMapMethod(
                                                   ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:99:5: Error: '_initializeFirebaseAppFromMap' is already declared in this scope.
        _initializeFirebaseAppFromMap(await channel.invokeMapMethod(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:40:8: Context: Previous declaration of '_initializeFirebaseAppFromMap'.
      void _initializeFirebaseAppFromMap(Map<dynamic, dynamic> map) {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:5: Error: Expected a class member, but got 'return'.
        return appInstances[name];
        ^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:12: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
    Try adding the name of the type of the variable or the keyword 'var'.
        return appInstances[name];
               ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:12: Error: Expected ';' after this.
        return appInstances[name];
               ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:12: Error: 'appInstances' is already declared in this scope.
        return appInstances[name];
               ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:11:48: Context: Previous declaration of 'appInstances'.
      static Map<String, MethodChannelFirebaseApp> appInstances = {};
                                                   ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:24: Error: Expected a class member, but got '['.
        return appInstances[name];
                           ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:25: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
    Try adding the name of the type of the variable or the keyword 'var'.
        return appInstances[name];
                            ^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:25: Error: Expected ';' after this.
        return appInstances[name];
                            ^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:29: Error: Expected a class member, but got ']'.
        return appInstances[name];
                                ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:104:30: Error: Expected a class member, but got ';'.
        return appInstances[name];
                                 ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:95:9: Error: 'appInstances.containsKey' can't be used as a type because 'appInstances' doesn't refer to an import prefix.
        if (appInstances.containsKey(name)) {
            ^^^^^^^^^^^^^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:119:1: Error: Expected a declaration, but got '}'.
    }
    ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:99:35: Error: Type 'await' not found.
        _initializeFirebaseAppFromMap(await channel.invokeMapMethod(
                                      ^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:8:7: Error: The non-abstract class 'MethodChannelFirebase' is missing implementations for these members:
     - MethodChannelFirebase._initializeFirebaseAppFromMap
    Try to either
     - provide an implementation,
     - inherit an implementation from a superclass or mixin,
     - mark the class as abstract, or
     - provide a 'noSuchMethod' implementation.

    class MethodChannelFirebase extends FirebasePlatform {
          ^^^^^^^^^^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:99:5: Context: 'MethodChannelFirebase._initializeFirebaseAppFromMap' is defined here.
        _initializeFirebaseAppFromMap(await channel.invokeMapMethod(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase_app.dart:55:27: Error: Can't use 'appInstances' because it is declared more than once.
        MethodChannelFirebase.appInstances.remove(name);
                              ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:34:18: Error: Can't use '_initializeFirebaseAppFromMap' because it is declared more than once.
        apps.forEach(_initializeFirebaseAppFromMap);
                     ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:48:27: Error: Can't use 'appInstances' because it is declared more than once.
        MethodChannelFirebase.appInstances[methodChannelFirebaseApp.name] =
                              ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:59:12: Error: Can't use 'appInstances' because it is declared more than once.
        return appInstances.values.toList(growable: false);
               ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:83:11: Error: Can't use 'appInstances' because it is declared more than once.
              appInstances[defaultFirebaseAppName];
              ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:88:14: Error: Can't use 'appInstances' because it is declared more than once.
          return appInstances[defaultFirebaseAppName];
                 ^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:99:35: Error: 'await' isn't a type.
        _initializeFirebaseAppFromMap(await channel.invokeMapMethod(
                                      ^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:113:9: Error: Getter not found: 'appInstances'.
        if (appInstances.containsKey(name)) {
            ^^^^^^^^^^^^
    ../../../.pub-cache/hosted/pub.dartlang.org/firebase_core_platform_interface-2.0.0/lib/src/method_channel/method_channel_firebase.dart:114:14: Error: Getter not found: 'appInstances'.
          return appInstances[name];
                 ^^^^^^^^^^^^

    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the precompiled application for the device.

Error launching application on Yong's iPhone.

```
Cleaning project or delete DerievedData and rerunning didn't work.
Am I missing/doing something wrong? It's occuring both Android Studio and VSCode.

### Expected behavior

Should build successfully.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
JYMAC:firebase_test_app 1100287$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.7 19H2 x86_64, locale en-KR)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.50.1)
[✓] Connected device (3 available)

```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
JYMAC:firebase_test_app 1100287$ flutter pub deps -- --style=compact
Dart SDK 2.11.0-213.1.beta
Flutter SDK 1.23.0-18.1.pre
firebase_test_app 1.0.0+1

dependencies:
- cupertino_icons 1.0.0
- firebase_core 0.5.0+1 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.5.0-nullsafety.1 [collection]
- boolean_selector 2.1.0-nullsafety.1 [source_span string_scanner]
- characters 1.1.0-nullsafety.3
- charcode 1.2.0-nullsafety.1
- clock 1.1.0-nullsafety.1
- collection 1.15.0-nullsafety.3
- fake_async 1.2.0-nullsafety.1 [clock collection]
- firebase 7.3.2 [http http_parser js]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http 0.12.2 [http_parser path pedantic]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- js 0.6.3-nullsafety.1
- matcher 0.12.10-nullsafety.1 [stack_trace]
- meta 1.3.0-nullsafety.4
- path 1.8.0-nullsafety.1
- pedantic 1.9.2 [meta]
- plugin_platform_interface 1.0.3 [meta]
- quiver 2.1.4+1 [matcher meta]
- sky_engine 0.0.99
- source_span 1.8.0-nullsafety.2 [charcode collection path term_glyph]
- stack_trace 1.10.0-nullsafety.2 [path]
- stream_channel 2.1.0-nullsafety.1 [async]
- string_scanner 1.1.0-nullsafety.1 [charcode source_span]
- term_glyph 1.2.0-nullsafety.1
- test_api 0.2.19-nullsafety.2 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0-nullsafety.3 [collection]
- vector_math 2.1.0-nullsafety.3

```

</details>

---

**Comment by Taewook-hotsunny** [comment]
Is there solution for this issue? I have facing same problem here

**Comment by markusaksli-nc** [comment]
Hi @DragonCherry @Taewook-hotsunny 
There is already an open issue regarding this. The problem seems to be that the `firebase_core_platform_interface.dart` goes missing for some reason. can you try https://github.com/FirebaseExtended/flutterfire/issues/2604 and see if it fixes the issue for you? You can comment on the mentioned issue with your results if it doesn't work.
Closing this as a duplicate of #2604

**Comment by DragonCherry** [comment]
Thank you for your response @markusaksli-nc ,
I found it fixed by commenting out flutter_core 0.5.0 and replace it 0.5.1.
Following file caused the issue
firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart
at line 81~88,
```dart
    if (name == null) {
      MethodChannelFirebaseApp defaultApp =
          appInstances[defaultFirebaseAppName];

        throw coreNotInitialized();
      }

      return appInstances[defaultFirebaseAppName];
    }
```
Obviously this code won't work since the braces not matched. It's been fixed as follow
```dart
    if (name == null) {
      MethodChannelFirebaseApp defaultApp =
          appInstances[defaultFirebaseAppName];

      if (defaultApp == null) {
        throw coreNotInitialized();
      }

      return appInstances[defaultFirebaseAppName];
    }
```
Now it's working well, I don't think I have to add downgraded firebase_core_platform_interface: ^1.0.2 anymore.
